src/Language/PureScript/Parser/Types.hs

```diff
@@ -126,3 +127,11 @@ parseRowEnding = P.option REmpty $ indented *> pipe *> indented *> parseType
 
 parseRow :: TokenParser Type
 parseRow = (curry rowFromList <$> commaSep (parseNameAndType parsePolyType) <*> parseRowEnding) P.<?> "row"
+
+parseConstraintArg :: TokenParser Type
+parseConstraintArg = indented *> P.choice [ P.try parseFunction
```

Comment:
I think we need `parseRow` here too.

But I'm wondering if we should just take `parseConstrainedType` out of `parseTypeAtom`, and try to parse `=>` like a type operator. Let me think about this a bit.